gcc/rust/typecheck/rust-casts.cc

```diff
@@ -63,11 +63,11 @@ TypeCastRules::cast_rules ()
 	      from.get_ty ()->debug_str ().c_str (),
 	      to.get_ty ()->debug_str ().c_str ());
 
-  switch (from.get_ty ()->get_kind ())
+  switch (from.get_ty ()->destructure ()->get_kind ())
```

Comment:
I think it would be better to do something like the unify code over in rust-unify.cc UnifyRules::go

```
TyTy::BaseType *
UnifyRules::go ()
{
  TyTy::BaseType *ltype = lhs.get_ty ();
  TyTy::BaseType *rtype = rhs.get_ty ();

  ltype = lhs.get_ty ()->destructure ();
  rtype = rhs.get_ty ()->destructure ();
```

So that you only need to call it once and it doesn't get lost.

Reply:
Edited it, please take another look.